lib/lab_manager/provider/vsphere.rb

```diff
@@ -313,5 +324,44 @@ def machine_present_in_drs_rule?(vs, group:, machine:, datacenter:)
       machine_short_name = machine.sub!(%r{^.*\/}, '')
       !(group.vm.find { |v| v.name == machine_short_name }).nil?
     end
+
+    def upload_file_impl(opts)
+      VSphere.connect.with do |vs|
+        conn = vs.instance_variable_get('@connection'.to_sym)
+
+        auth = RbVmomi::VIM::NamePasswordAuthentication(
+          username: opts[:user],
+          password: opts[:password],
+          interactiveSession: false
+        )
+
+        file_manager = conn.serviceContent.guestOperationsManager.fileManager
+
+        Retryable.retryable(tries: 3, exception_cb: RETRYABLE_CALLBACK) do
+          endpoint = file_manager.InitiateFileTransferToGuest(
+            vm: vm_data['mo_ref'],
+            auth: auth,
+            guestFilePath: opts[:guestFilePath],
+            overwrite: opts[:overwrite] || false,
+            fileAttributes: RbVmomi::VIM::GuestWindowsFileAttributes.new,
+            fileSize: opts[:hostFile].size
+          )
+
+          uri = URI.parse(endpoint)
+          http = Net::HTTP.new(uri.hostname, uri.port )
+          http.use_ssl = true
+          http.verify_mode = OpenSSL::SSL::VERIFY_NONE
```

Comment:
hardcoded? why not verify ssl? 

Could you make it configurable? It's connection specific, so I don't know the right place where to configure it:(